x/upgrade/keeper/keeper.go

```diff
@@ -326,22 +326,28 @@ func (k Keeper) IsSkipHeight(height int64) bool {
 }
 
 // DumpUpgradeInfoToDisk writes upgrade information to UpgradeInfoFileName.
-func (k Keeper) DumpUpgradeInfoToDisk(height int64, name string) error {
+// `info` should be provided and contain Plan.Info data in order to support
+// auto upgrade functionality by cosmovisor and other tools using upgarde-info.json
+// (GetUpgradeInfoPath()) file.
+func (k Keeper) DumpUpgradeInfoToDisk(height int64, name string, info ...string) error {
```

Comment:
trick to make the API backward compatible.

Reply:
Why don't we just add another overload `DumpFullUpgradeInfoToDisk`? The varargs feels too hacky.

Reply:
The `master` branch already has it as `DumpUpgradeInfoToDisk(height int64, p types.Plan) error`. This PR is just to get the functionality into `v0.44.x`.

Reply:
The question was if we should do it by adding new function (and instantly marking it "deprecated") or overloading the existing `DumpUpgradeInfoToDisk`.
In general overloading smells bad, so Aaron comment makes sense.